src/main/java/org/jabref/gui/collab/ExternalChangesResolverViewModel.java

```diff
@@ -82,10 +86,18 @@ public void acceptChange() {
             selectedChange.accept();
             getVisibleChanges().remove(selectedChange);
         });
+        if (activeTab != null) {
+            activeTab.get().get().updateTabTitle(false);
+            activeTab.get().get().resetChangedProperties();
+        }
```

Comment:
I think, this is a **single** change.

The issue is about **all** changes. If **all** changes are accepted, there is no change. In case there is a single change denied, then the database should be changed.

In your code, if I deny and then accept, the asterisk is removed even if it should stay.

I think, the logic should go somewhere elese. It feels to deep in this ViewModel. Isn't it?

I think the detection should go into `org.jabref.gui.collab.ExternalChangesResolverViewModel#applyChanges`.

Reply:
logic moved to apply changes